C/include/simplicity/errorCodes.h

```diff
@@ -79,8 +79,8 @@ static inline const char * SIMPLICITY_ERR_MSG(simplicity_err err) {
     return "Expression not unit to unit";
   case SIMPLICITY_ERR_WITNESS_EOF:
     return "Unexpected end of witness block";
-  case SIMPLICITY_ERR_WITNESS_UNUSED_BITS:
-    return "Unused data at the end of the witness block";
+  case SIMPLICITY_ERR_WITNESS_LENGTH_MISMATCH:
+    return "Witness block is longer than declared";
```

Comment:
It isn't longer than declared.  The prefix of the witness block declares how long it is and the decoded block is always that long.  However the witness block contains a sequence of witness values, and it could be that the witness block is longer than necessary, in which case this error is raised.  It could also be shorter than necessary, in which case a `SIMPLICITY_ERR_WITNESS_EOF` is raised.

Reply:
I see. We always decode as many bits as declared at the start of the witness block. _(The maximum length is bounded by 2^31 exclusive to prevent DDoS.)_ If there are not enough bits, then we throw an EOF error. After parsing the witness, we check if there are trailing bits left, in which case we throw UNUSED_BITS.